File: src/main/scala/com/example/rest/MySecurityContext.java
package com.example.rest;

import java.security.Principal;
import java.util.List;

import javax.ws.rs.core.SecurityContext;

public class MySecurityContext implements SecurityContext {
	private final MyUser user;
	private final String scheme;
	private final List<String> roles;
	
	public MySecurityContext(MyUser user, String scheme, List<String> roles) {
		this.user = user;
		this.scheme = scheme;
		this.roles = roles;
	}

	@Override
	public Principal getUserPrincipal() {
		return user;
	}

	@Override
	public boolean isUserInRole(String role) {
		return roles.contains(role);
	}

	@Override
	public boolean isSecure() {
		return "https".equalsIgnoreCase(this.scheme);
	}

	@Override
	public String getAuthenticationScheme() {
		return SecurityContext.BASIC_AUTH;
	}
}

File: src/main/scala/com/example/rest/MySecurityContext.scala
package com.example.rest

import javax.ws.rs.core.SecurityContext

class MySecurityContext(private val user: MyUser, private val scheme: String, private val roles: List[String])
  extends SecurityContext {
  override def getUserPrincipal() = user

  override def isUserInRole(role: String) = roles.contains(role)

  override def isSecure() = "https".equalsIgnoreCase(scheme)

  override def getAuthenticationScheme() = SecurityContext.BASIC_AUTH
}
